### PR TITLE
ci(connect): start running api tests with the latest released fw in t…

### PR DESCRIPTION
**Pull request by mroz22**
before nightly connect api tests were run with firmware:
- canary (main branch fw repo)
- legacy (pinned older fw).

after: 
- canary (main branch fw repo)
- latest (currently released fw)
- legacy (pinned older fw)

run here https://github.com/trezor/trezor-suite/actions/runs/11724519643

### CODE REVIEW

File: .github/workflows/test-connect.yml [hunk 1]
```diff
@@ -83,6 +83,10 @@ jobs:
         id: set-matrix-legacy-firmware
         run: echo "legacyFirmwareMatrix=$(node ./scripts/ci/connect-test-matrix-generator.js --model=T2T1 --firmware=2.3.0 --env=all --groups=all --cache_tx=false --transport=Bridge)" >> $GITHUB_OUTPUT
 
+      - name: Set latest devices matrix
+        id: set-matrix-latest-firmware
+        run: echo "latestFirmware=$(node ./scripts/ci/connect-test-matrix-generator.js --model=T2T1 --firmware=2-latest --env=all --groups=all --cache_tx=false --transport=Bridge)" >> $GITHUB_OUTPUT
```
Comment (martykan) [inline .github/workflows/test-connect.yml]:
```suggestion
        run: echo "latestFirmwareMatrix=$(node ./scripts/ci/connect-test-matrix-generator.js --model=T2T1 --firmware=2-latest --env=all --groups=all --cache_tx=false --transport=Bridge)" >> $GITHUB_OUTPUT
```
Variable name doesn't match

**Comment by mroz22** [comment]
less CI boilerplate. what do  you think? 🤔  https://github.com/trezor/trezor-suite/actions/runs/11780131566

File: .github/workflows/test-connect.yml [hunk 2]
```diff
@@ -128,41 +123,23 @@ jobs:
...
-      testFirmwareModel: ${{ matrix.model }}
-    strategy:
-      fail-fast: false
-      matrix: ${{ fromJson(needs.set-matrix.outputs.legacyFirmwareMatrix) }}
-
-  canary-fw:
+  all-fws:
     needs: [build, set-matrix]
-    name: canary-fw ${{ matrix.key }}
+    name: all-fws ${{ matrix.key }}
```
Comment (martykan) [inline .github/workflows/test-connect.yml]: Maybe we could put `${{ matrix.firmware }}` here so it's grouped

Comment (mroz22) [reply .github/workflows/test-connect.yml]: let me try

Comment (mroz22) [reply .github/workflows/test-connect.yml]: something like this? https://github.com/trezor/trezor-suite/actions/runs/11780696955

**Review by martykan — APPROVED** [review]

**Pull request merged**